### ISSUE TITLE
Any good reason to disallow identifiers beginning with digit?

### ISSUE THREAD

**Issue by ds26gte** [brownplt/pyret-lang#1764]
Also identifiers starting with hyphen.  These would be great for naming the results of algebraic operations, such as
```
2x = 2 * x
-y = 0 - y
```

**Comment by jpolitz** [comment]
Both of those are super misleading as “just” identifiers in algebra notation, and are things a student or teacher might write expecting an operation to happen.

I think this is a good nudge to give special error messages for those cases!

**Comment by shriram** [comment]
I agree. If we make these identifiers work, teachers will expect all kinds of things to "just work", especially the math teachers: if I write
```
x = 2
y = 4
2xy
```
is that `16`?

There are a handful of times in Racket where I make use of this, and it's true that I do miss it in Pyret in those situations. But the likelihood of confusion/misconceptions, especially with one of our major audiences (programming-unsophisticated middle-school math teachers) is far, far greater and more important.
